File: resources/c-sharp/case-statements.cs
double z = 0.0;
int a = 3;
int b = 5;
switch (z) {
    case < 0.0:
        break;
    case > 15.0:
        break;
    case 0:
        goto case double.NaN;
    case double.NaN:
        break;
    case (== 7.0) when a == b: //case guard
        break;
    default:
        break;
}

File: resources/kotlin/case-statements.kt
fun main() {
    val x = 3

    when (x) {
        1 -> print("x == 1")
        2 -> print("x == 2")
        3, 4, 5 -> print("x is 3, 4 or 5")
        else -> {
            print("x is mysterious")
        }
    }
}
